src/main/steps/applicant1/finalising-your-application/post.ts

```diff
@@ -19,7 +29,37 @@ export default class FinalisingYourApplicationPostController extends PostControl
     return super.save(req, formData, eventName);
   }
 
+  protected getNextUrl(req: AppRequest): string {
+    const hasApplicant2SwitchedToSoleFo =
+      req.session.isApplicant2 &&
+      req.originalUrl === APPLICANT_2 + FINALISING_YOUR_APPLICATION &&
+      req.session.userCase.finalOrderSwitchedToSole === YesOrNo.YES &&
+      this.formData.doesApplicant2WantToApplyForFinalOrder;
+
+    if (hasApplicant2SwitchedToSoleFo) {
+      req.session.isApplicant2 = false;
```

Comment:
not sure this is the right place to be changing this flag, because it has wider implications apart from the nextUrl
I think it should probably be updated when the event to switch to sole is submitted
there might be other things in the session that also need to be updated as a result of that

Reply:
We shouldn't change it on submit - we should only change it if the switch to sole event was successful.  At that point the applicant data has been entirely swapped, and we should be switching the flag.  I'm not sure if anything else should also be updated - it's possible, but at this point the end user journey is complete and they will only have the option to logout.  I believe the risk is negligible.

Reply:
I realised that an error count check was missing from hasApplicant2SwitchToSoleFo - this has been corrected.